From 13418541a5ae19b15f51cbb87faf344902f5af98 Mon Sep 17 00:00:00 2001
From: Chris Liddell <chris.liddell@artifex.com>
Date: Thu, 6 Sep 2018 18:40:05 +0100
Subject: [PATCH] Bug 699722 (2): add wildcards to the permissions paths.

The temp and ICC profile paths need to finish with wildcards to work correctly.
---
 Resource/Init/gs_init.ps | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Resource/Init/gs_init.ps b/Resource/Init/gs_init.ps
index 83918a9..5ff9f63 100644
--- Resource/Init/gs_init.ps.orig
+++ Resource/Init/gs_init.ps
@@ -2034,7 +2034,7 @@ readonly def
     <<
       /PermitFileReading [
         currentuserparams /PermitFileReading get aload pop
-        //tempfilepaths aload pop
+        [//tempfilepaths aload pop] (*) .generate_dir_list_templates
         /FONTPATH .systemvar (*) .generate_dir_list_templates
           % Library files :
         /LIBPATH  .systemvar (*) .generate_dir_list_templates
@@ -2049,15 +2049,15 @@ readonly def
           [ currentsystemparams /GenericResourceDir get] (*) .file_name_separator (*)
             concatstrings concatstrings .generate_dir_list_templates
         } if
-        currentuserparams /ICCProfilesDir known {currentuserparams /ICCProfilesDir get} if
+        currentuserparams /ICCProfilesDir known {[currentuserparams /ICCProfilesDir get] (*) .generate_dir_list_templates} if
       ]
       /PermitFileWriting [
           currentuserparams /PermitFileWriting get aload pop
-          //tempfilepaths aload pop
+          [//tempfilepaths aload pop] (*) .generate_dir_list_templates
       ]
       /PermitFileControl [
           currentuserparams /PermitFileControl get aload pop
-          //tempfilepaths aload pop
+          [//tempfilepaths aload pop] (*) .generate_dir_list_templates
       ]
       /LockFilePermissions //true
     >> setuserparams
@@ -2140,7 +2140,7 @@ SAFER { .setsafeglobal } if
   /.setshapealpha            % transparency-example.ps
   /.endtransparencygroup     % transparency-example.ps
   /.setdotlength             % Bug687720.ps
-  /.sort /.setdebug /.mementolistnewblocks /getenv
+  /.sort /.mementolistnewblocks /getenv
 
   /.makeoperator /.setCPSImode              % gs_cet.ps, this won't work on cluster with -dSAFER
 
-- 
2.9.1

